powerdnsadmin/routes/domain.py

```diff
@@ -646,6 +691,43 @@ def change_account(domain_name):
         abort(500)
 
 
+def are_equal(old : RecordEntry, new: dict, domain_name: str):
+    if old.name == domain_name:
+        name_comparison = (new['record_name'] == '@')
+    else:
+        name_comparison = (old.name == new['record_name'] + "." + domain_name)
+
+    if not name_comparison or \
+            old.type != new['record_type'] or \
+            old.status != new['record_status'] or \
+            int(old.ttl) != int(new['record_ttl']) or \
+            old.data != new['record_data'] or \
+            old.comment != new['record_comment']:
+        return False
+    return True
+
+def get_changed_record_entries(old_entries, new_submission, current_user, domain_name):
+    not_changed = []
```

Comment:
I think this portion of code could be rewritten in a single loop. Plus the json dumps is overkill if you can identify an object with one of it's property.